feat: platform onboarding

JrmyDev opened the pull request:
#760 

- [x] llm proxy
- [x] agent creation
- [x] mcp gateway 
- [x] tool assignment
- [x] oauth management
- [ ] redirect directly to tools assign after oauth callback (currently not smooth because it reload onboarding from step 0 then load tools assign step)

iskhakov commented on platform/frontend/src/components/onboarding/onboarding-wizard.tsx:
```diff
@@ -0,0 +1,353 @@
+                    : f === "openwebui"
+                      ? "OpenWebUI"
+                      : "Custom"}
+              </OptionButton>
```
Please create the variable with the frameworks so it is not hardcoded throughout the code

iskhakov commented on platform/frontend/src/components/ui/onboarding-step.tsx:
```diff
@@ -0,0 +1,76 @@
+import type { ReactNode } from "react";
+
```
Is this a default shadcn component? Do we only add default shadcn components to this folder, @brojd?

brojd commented on platform/frontend/src/components/onboarding/onboarding-wizard.tsx:
```diff
@@ -0,0 +1,351 @@
+  const [framework, setFramework] = useState<"n8n" | "langchain" | "openwebui">(
+    "n8n",
+  );
+  const [provider, setProvider] = useState<"openai" | "anthropic">("openai");
```
Those + frameworks should not be hardcoded this way. If we have existing type for it (like SupportedProvidersInput in frontend/src/lib/clients/api/types.gen.ts) let's use it. 

If sth is missing like the frameworks, I suggest to put it in shared package as a const / enum and reuse

JrmyDev replied:
Yep I add type framework and forgot replacing here

brojd commented on platform/frontend/src/app/globals.css:
```diff
@@ -199,3 +199,19 @@ button, div[role="menuitem"] {
+  max-width: 100%;
+  height: auto;
+  object-fit: contain;
+}
```
Are those used anywhere?

JrmyDev commented on platform/frontend/src/components/ui/copy-button.tsx:
```diff
@@ -0,0 +1,34 @@
+  }
+
+  return (
+    <button
```
I might have mess with my code because I had previously replace many.
Anyway I will change all with shadnc :)

JrmyDev commented on platform/frontend/src/lib/constant.ts:
@brojd i need to access getDisplayProxyUrl that's why i put some constants here, what do you think?

JrmyDev commented:
> jsut curious why it shows 920 tools identified?

in drizzle ? no idea

brojd commented:
> jsut curious why it shows 920 tools identified?

I suppose that's the result of hitting `db-seed-mock-data` in tilt many times

JrmyDev commented:
hmm i didn't use much db-seed-mock-data in tilt, 
in drizzle, count on left is not matching table rows no? 876 vs 48
<img width="2380" height="1287" alt="image" src="https://github.com/user-attachments/assets/7fdccfb5-5b6d-460e-8cd8-135af36ef907" />

JrmyDev commented:
> @JrmyDev please let us know when you address this bug and we can merge the pr

yes i will check

JrmyDev commented:
> This week we added mcp server gateway to the platform. So we need some additional steps to the onboarding:
> 
> 1. Explain that archestra acts as 1. LLM proxy and 2. MCP Gateway
> 2. Ask how user wants to use it with multiselect
> 3. Create Agent
> 4. If MCP Gateway is selected -> Ask to add MCP server from External Registry
> 5. Install that MCP server
> 6. Assign tools from MCP server to Agent
> 7. Provide the link to Agent's MCP gateway and mcp.json snippet so users can add it to their mcp clients (cursor, claude, ...)
> 8. if LLM proxy is selected -> instructions how to connect agent to LLM proxy that we already implemented

@iskhakov i almost finish adding this flow (maybe last step missing)
rush a bit before being busy so might be some code refactor
